src/analysis/powerset-lattice-impl.h

```diff
@@ -5,61 +5,81 @@
-  return element.value.all();
-}
+  // True in left, false in right.
+  bool leftNotRight = false;
 
-template<size_t N>
-inline bool
-BitsetPowersetLattice<N>::isBottom(const BitsetPowersetLattice<N>& element) {
-  // Bottom lattice element is the empty set.
-  return element.value.none();
-}
+  // True in right, false in left.
+  bool rightNotLeft = false;
 
-template<size_t N>
-inline LatticeComparison
-BitsetPowersetLattice<N>::compare(const BitsetPowersetLattice<N>& left,
-                                  const BitsetPowersetLattice<N>& right) {
-  size_t leftCount = left.value.count();
-  size_t rightCount = right.value.count();
-
-  // If left has more elements, left might be a superset of right.
-  if (leftCount > rightCount) {
-    if ((left.value | right.value) == left.value) {
-      return GREATER;
+  size_t size = left.bitvector.size();
+
+  for (size_t i = 0; i < size; ++i) {
+    leftNotRight |= (left.bitvector[i] && !right.bitvector[i]);
+    rightNotLeft |= (right.bitvector[i] && !left.bitvector[i]);
+
+    // We can end early if we know neither is a subset of the other.
+    if (leftNotRight && rightNotLeft) {
+      return NO_RELATION;
     }
-    // If right has more elements, right might be a superset of left.
-  } else if (leftCount < rightCount) {
-    if ((left.value | right.value) == right.value) {
-      return LESS;
+  }
+
+  if (!leftNotRight) {
+    if (!rightNotLeft) {
+      return EQUAL;
     }
-  } else if (left.value == right.value) {
-    return EQUAL;
+    return LESS;
+  } else if (!rightNotLeft) {
+    return GREATER;
   }
 
   return NO_RELATION;
 }
 
-// Implement LUB as an OR of the two bitsets.
-template<size_t N>
-inline void BitsetPowersetLattice<N>::getLeastUpperBound(
-  const BitsetPowersetLattice<N>& right) {
-  value |= right.value;
+inline FinitePowersetLattice::Element FinitePowersetLattice::getBottom() {
+  FinitePowersetLattice::Element result(setSize);
+  return result;
+}
+
+// We count the number of element members present in the element by counting the
+// trues in the bitvector.
+inline size_t FinitePowersetLattice::Element::count() {
+  size_t count = 0;
+  for (auto it : bitvector) {
+    count += it;
+  }
+  return count;
+}
+
+// Least upper bound is implemented as a logical OR between the bitvectors on
+// both sides. We return true if a bit is flipped in-place on the left so the
+// worklist algorithm will know if when to enqueue more work.
+inline bool FinitePowersetLattice::Element::makeLeastUpperBound(
+  const FinitePowersetLattice::Element& right) {
+  // Both must be from powerset lattice of the same set.
+  assert(right.bitvector.size() == bitvector.size());
+
+  bool modified = false;
+  for (size_t i = 0; i < bitvector.size(); ++i) {
+    // Bit is flipped on left only if left is false and right is true when left
+    // and right are OR'ed together.
+    modified |= (!bitvector[i] && right.bitvector[i]);
+    bitvector[i] = bitvector[i] || right.bitvector[i];
```

Comment:
```suggestion
    bitvector[i] = bitvector[i] | right.bitvector[i];
```
It's almost certainly optimized by the compiler, but still, we just want a logical OR here and not control flow branching.

Reply:
@kripken The CI does not like the use of bitwise operands with booleans unless cast to an int:
`error: use of bitwise '|' with boolean operands [clang-diagnostic-bitwise-instead-of-logical]`.

Reply:
Probably not worth working around. I would keep using `||`.

Reply:
Interesting. More stuff about C++ I did not know... `||` sgtm then